### PR TITLE
support job metrics command

### PR DESCRIPTION
**Pull request by kdh0102**
This PR adds `job metrics` commands: `metrics list` and `metrics show`

### Example
The following is an example of using job metrics commands:

```bash
> pf job metrics list <JOB_ID>
        Metrics List
┏━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Name                     ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ train_loss               │
└──────────────────────────┘
```

```bash
# NOTE that you can pass multiple metrics names.
> pf job metrics show <JOB_ID> --name <METRICS_NAME_1>
                               Metrics
┏━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┓
┃ Name       ┃ Created                          ┃ Value             ┃
┡━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
│ train_loss │ 2022-09-24T14:27:19.748559+00:00 │ 7.088735595703125 │
└────────────┴──────────────────────────────────┴───────────────────┘
```
### Minor Changes
To pass additional header keywords I have made modifications in `client.base`.
Please have a look and let me know if there is any issue.

### CODE REVIEW

File: pfcli/job.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 from datetime import datetime
 
 import tabulate
+from pfcli.service.client.metrics import MetricsClientService
```
Comment (ktaebum) [inline pfcli/job.py:14]: group pfcli imports

File: pfcli/job.py [hunk 2]
```diff
@@ -602,3 +619,25 @@ def log(
...
+
+@metrics_app.command("show")
+def show_metrics(
+    job_id: int = typer.Argument(..., help="ID of job"),
+    name: List[str] = typer.Option(..., help="metrics name"),
+):
+    """Show metrics values"""
+    client: MetricsClientService = build_client(ServiceType.METRICS)
+    for metrics in name:
+        results = client.get_metrics_values(job_id, metrics)
```
Comment (ktaebum) [inline pfcli/job.py:642]:
```
for metric_name in name:
    metrics = client.get_metrics_values(job_id, metric_name)
```
?

File: pfcli/service/client/__init__.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 from pfcli.service.client.base import ClientService
 from pfcli.service.client.billing import BillingClientService
 from pfcli.service.client.checkpoint import CheckpointClientService
+from pfcli.service.client.metrics import MetricsClientService
```
Comment (ktaebum) [inline pfcli/service/client/__init__.py:10]: sort? (alphabetical order)
after `pfcli.service.client.job`

File: pfcli/service/client/metrics.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        query = "{ sampleQuery(fields: {" + fields + "}) {name} }"
+        resp = safe_request(
+            self.post, err_prefix=f"Failed to get metrics of ({job_id})."
+        )(
+            headers={"Content-Type": "application/json"},
+            data=json.dumps({"query": query}),
+        )
+        data = resp.json()["data"]["sampleQuery"]
+        names = set(sample.get("name", None) for sample in data)
+        return [{"name": name} for name in names if name]
```
Comment (ktaebum) [inline pfcli/service/client/metrics.py:29]: This function does not return `List[str]`, but `List[Dict[str, str]]`. Just returning `[name for name in names if name]` seems okay.

File: pfcli/service/client/metrics.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    def get_metrics_values(
+        self,
+        job_id: int,
+        name: str,
+        limit: int = 10,
+    ) -> dict[str, str]:
```
Comment (ktaebum) [inline pfcli/service/client/metrics.py:36]: `List[Dict[str, Any]]`? If we have type constraint for metric value, we can define Type variable `MetricType`.

File: pfcli/service/client/metrics.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        fields = (
+            f'task: "{job_id}", telemetry: "{name}", limit: {limit}, goal: LAST'
+        )
+        query = "{ representativeSamples(fields: {" + fields + "}) {created, value} }"
+        resp = safe_request(
+            self.post, err_prefix=f"Failed to get metrics of ({job_id})."
+        )(
+            headers={"Content-Type": "application/json"},
+            data=json.dumps({"query": query}),
+        )
```
Comment (ktaebum) [inline pfcli/service/client/metrics.py:46]: Maybe we need to keep retrieving the results? (`limit` denotes pagination?)

**Review by ktaebum — COMMENTED** [review]
Thx. just left some comments

File: pfcli/service/client/metrics.py (hunk 6 above)
Comment (kdh0102) [reply pfcli/service/client/metrics.py:46]: At the moment, we only support showing the latest N values.
`limit` specifies maximum values to retrieve.

**Comment by kdh0102** [comment]
@ktaebum Thanks for your valuable comments! I've addressed all your comments.

**Review by kooyunmo — CHANGES_REQUESTED** [review]
As requested offline, please add the following changes:

1. Adding an "iteration" column
2. Making the timestamp in the human-readable format
3. Rounding the floating point values

**Comment by kdh0102** [comment]
@kooyunmo  Thanks for your comments. Please check the changes.

Now `metrics show` command will show results as follows:
```bash
                     Metrics
┏━━━━━━┳━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┓
┃ Name ┃ Iteration ┃ Created             ┃ Value ┃
┡━━━━━━╇━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━┩
│ loss │ 9800      │ 2022-09-26 12:01:11 │ 1.757 │
│ loss │ 9799      │ 2022-09-26 12:01:10 │ 2.32  │
│ loss │ 9798      │ 2022-09-26 12:01:10 │ 2.349 │
│ loss │ 9797      │ 2022-09-26 12:01:10 │ 2.202 │
│ loss │ 9796      │ 2022-09-26 12:01:10 │ 2.113 │
│ loss │ 9795      │ 2022-09-26 12:01:10 │ 2.215 │
│ loss │ 9794      │ 2022-09-26 12:01:10 │ 2.301 │
│ loss │ 9793      │ 2022-09-26 12:01:10 │ 2.248 │
│ loss │ 9792      │ 2022-09-26 12:01:09 │ 2.433 │
│ loss │ 9791      │ 2022-09-26 12:01:09 │ 1.96  │
└──────┴───────────┴─────────────────────┴───────┘
```

**Review by kooyunmo — APPROVED** [review]
LGTM

**Pull request merged**